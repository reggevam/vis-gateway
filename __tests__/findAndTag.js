const findAndTag = require('./../src/utils/findAndTag');
const {
  setupHighlightArray,
  findAndTag: findAndTagWorker,
} = require('./../src/workers/index');

describe('findAndTag', () => {
  const content = 'the big brown fox jumped over the fence';
  const tags = [
    { text: 'the', confidence: 0.1 },
    { text: 'big', confidence: 0.2 },
    { text: 'over', confidence: 0.3 },
    { text: 'fence', confidence: 0.4 },
  ];
  it('should construct and offsets array', () => {
    expect(findAndTag(content, tags, 'text')).toMatchSnapshot();
  });

  it('should run async as a worker', async () => {
    expect(await findAndTagWorker(content, tags, 'text')).toMatchSnapshot();
  });

  it('should construct a highlight array from the offset array', async () => {
    const offsetArray = findAndTag(content, tags, 'text');
    const highlightArray = await setupHighlightArray(content, offsetArray);
    expect(highlightArray).toMatchSnapshot();
  });
});
