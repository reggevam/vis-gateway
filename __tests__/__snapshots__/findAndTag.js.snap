// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findAndTag should construct a highlight array from the offset array 1`] = `
Array [
  Object {
    "isEntity": false,
    "text": "",
  },
  Object {
    "confidence": 0.1,
    "endOffset": 3,
    "isEntity": true,
    "startOffset": 0,
    "text": "the",
  },
  Object {
    "isEntity": false,
    "text": " ",
  },
  Object {
    "confidence": 0.2,
    "endOffset": 7,
    "isEntity": true,
    "startOffset": 4,
    "text": "big",
  },
  Object {
    "isEntity": false,
    "text": " brown fox jumped ",
  },
  Object {
    "confidence": 0.3,
    "endOffset": 29,
    "isEntity": true,
    "startOffset": 25,
    "text": "over",
  },
  Object {
    "isEntity": false,
    "text": " the ",
  },
  Object {
    "confidence": 0.4,
    "endOffset": 39,
    "isEntity": true,
    "startOffset": 34,
    "text": "fence",
  },
  Object {
    "isEntity": false,
    "text": "",
  },
]
`;

exports[`findAndTag should construct and offsets array 1`] = `
Array [
  Object {
    "confidence": 0.1,
    "endOffset": 3,
    "startOffset": 0,
    "text": "the",
  },
  Object {
    "confidence": 0.2,
    "endOffset": 7,
    "startOffset": 4,
    "text": "big",
  },
  Object {
    "confidence": 0.3,
    "endOffset": 29,
    "startOffset": 25,
    "text": "over",
  },
  Object {
    "confidence": 0.4,
    "endOffset": 39,
    "startOffset": 34,
    "text": "fence",
  },
]
`;

exports[`findAndTag should run async as a worker 1`] = `
Array [
  Object {
    "confidence": 0.1,
    "endOffset": 3,
    "startOffset": 0,
    "text": "the",
  },
  Object {
    "confidence": 0.2,
    "endOffset": 7,
    "startOffset": 4,
    "text": "big",
  },
  Object {
    "confidence": 0.3,
    "endOffset": 29,
    "startOffset": 25,
    "text": "over",
  },
  Object {
    "confidence": 0.4,
    "endOffset": 39,
    "startOffset": 34,
    "text": "fence",
  },
]
`;
